client/my-sites/preview/index.js

```diff
@@ -9,6 +9,6 @@ import { siteSelection, sites, navigation } from 'my-sites/controller';
 import { preview } from './controller';
 
 export default function( router ) {
-	router( '/view', siteSelection, sites );
+	router( '/view', siteSelection, sites, makeLayout );
```

Comment:
Late to the party :slightly_frowning_face: IIRC having both `siteSelection` and `sites` on the same route def wasn't quite right, since you'd either instruct Calypso to extract a site slug from the current URL and set the currently selected site (single-site) mode (`sitesSelection`), _or_, in multi-site mode, you'd open the site selector (`sites`). So in this case (multi-site route), I think we don't need `siteSelection`.

Reply:
Yay, sharp eyes @ockham!

There is a difference for _users without sites_.

With `siteSelection` all good:

![screen shot 2018-01-25 at 12 14 18](https://user-images.githubusercontent.com/87168/35382929-689dac98-01c9-11e8-9b9e-a65b7b3874f0.png)

Without `siteSelection` we get unfriendly dead end:

![screen shot 2018-01-25 at 12 14 10](https://user-images.githubusercontent.com/87168/35382935-6acd2e76-01c9-11e8-98dd-38831b8386ce.png)

For users with sites on their list, there's no difference.

Seems like site selector should show that nicer screen in this situation, though! Right now it's a bit of a dead end for what looks like 7 routes in the codebase:
- `/extensions/wp-job-manager`
- `/extensions/wp-super-cache`
- `/extensions/zoninator`
- `/extensions/zoninator/new`
- `/extensions/zoninator/zone`
- `/plugins/upload`
- `/themes/upload`

Other 11 routes have both `siteSelection` and `sites`.

I had a quick look at [`site-selector`](https://github.com/Automattic/wp-calypso/blob/90e00bd4949a4a92fd75ecb4eae00f0e66986ba1/client/components/site-selector/index.jsx) component and it gets a bit complicated with all the hidden sites and stuff. Can you see an easy place there to add [`no-sites-message`](https://github.com/Automattic/wp-calypso/blob/90e00bd4949a4a92fd75ecb4eae00f0e66986ba1/client/components/empty-content/no-sites-message/index.jsx) component? 
An easy but semantically incorrect way would be to just add `siteSelection` everywhere. ;-)

Reply:
> I had a quick look at [`site-selector`](https://github.com/Automattic/wp-calypso/blob/90e00bd4949a4a92fd75ecb4eae00f0e66986ba1/client/components/site-selector/index.jsx) component and it gets a bit complicated with all the hidden sites and stuff. Can you see an easy place there to add [`no-sites-message`](https://github.com/Automattic/wp-calypso/blob/90e00bd4949a4a92fd75ecb4eae00f0e66986ba1/client/components/empty-content/no-sites-message/index.jsx) component? 

Aight, I had another stab at this and made a PR (#21911); sorry @ockham for earlier asking you to basically something I should've done myself! ;-) I'm just very intimidated to touch any site-selector code without being more familiar with it. It's one of those complicated card-house-like parts of Calypso with tons of exceptions hidden deep inside it... even small changes in it seem to always break something.

Looks like `siteSelector` does quite a lot and is needed when a user has just one site or no sites at all. So it has to always be there together with `sites`. 🤷‍♂️

Reply:
> Aight, I had another stab at this and made a PR (#21911); sorry @ockham for earlier asking you to basically something I should've done myself! ;-) I'm just very intimidated to touch any site-selector code without being more familiar with it. It's one of those complicated card-house-like parts of Calypso with tons of exceptions hidden deep inside it... even small changes in it seem to always break something.

No worries, really appreciate your work on this! Sry for being slow to reply, I'm a bit in the weeds of the JPO project and a team meetup right now :slightly_smiling_face: 

And I wholeheartedly agree, `siteSelector` is a pretty big and complex beast.